Stripe/PublicHeaders/STPCustomerContext.h

```diff
@@ -0,0 +1,67 @@
+//
+//  STPCustomerContext.h
+//  Stripe
+//
+//  Created by Ben Guo on 5/2/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "STPBackendAPIAdapter.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol STPEphemeralKeyProvider;
+@class STPEphemeralKey, STPEphemeralKeyManager;
+
+/**
+ An `STPCustomerContext` retrieves and updates a Stripe customer using
+ an ephemeral key, a short-lived API key scoped to a specific Customer object.
+ */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated"
+@interface STPCustomerContext : NSObject <STPBackendAPIAdapter>
+#pragma clang diagnostic pop
+
+/**
+ This is a convenience singleton CustomerContext. Before using the singleton
+ instance, you'll need to give it a key provider using `setKeyProvider`.
+ */
++ (instancetype)sharedInstance;
```

Comment:
What is the reasoning behind providing this singleton  given they have to set the key provider anyway, instead of just making them use the standard init?